### PR TITLE
Add manage_ruby, making the management of ruby in reports optional

### PR DESCRIPTION
**Pull request by downsj2**
Many people do not use the ruby module to manage ruby installs.  Invoking it, or attempting to manage other ruby packages directly, can and does break many things-- and the very last thing one wants to break on their puppet master is _ruby_.

This adds a manage_ruby option to the module, defaulting to true, which can then be set to false in order to prevent such carnage.

### CODE REVIEW

**Comment by truthbk** [comment]
Sounds more than fair @downsj2, sorry if this indeed broke things on your end 😞 

Thanks for keeping backward compatibility with the flag value. Could you please fix the spec tests before we merge? Thanks a lot for your contribution!

File: manifests/reports.pp [hunk 1]
```diff
@@ -15,27 +15,30 @@
 #
 class datadog_agent::reports(
   $api_key,
+  $manage_ruby,
```
Comment (truthbk) [inline manifests/reports.pp]: I think this should be set to `true` by default to avoid breaking manifests not accounting for the option.

Comment (truthbk) [inline manifests/reports.pp]: It will also fix the failing spec tests.

**Comment by truthbk** [comment]
I was testing some things out so I created a branch with this commit to identify/fix the issues - 7c8603e4de975731def26ff351b667a679b341a1

Feel free to cherry-pick! Alternatively, I could open a superseding PR (but I would rather merge yours in 😄 )

**Comment by albertvaka** [comment]
Rebased here: https://github.com/DataDog/puppet-datadog-agent/pull/613

**Pull request closed without merging**